File: web/src/api/smartlockApi.js
import api from './axios';

// ── Auth ─────────────────────────────────────────────────────────────────────
export const login = (email, password) =>
  api.post('/api/auth/login', { email, password });

export const register = (email, password, fullName) =>
  api.post('/api/auth/register', { email, password, fullName });

// ── Locks ─────────────────────────────────────────────────────────────────────
export const getAllLocks = () => api.get('/api/locks');

export const createLock = (name, serialNumber, timezone) =>
  api.post('/api/locks', { name, serialNumber, timezone });

export const deleteLock = (lockId) => api.delete(`/api/locks/${lockId}`);

export const editLock = (lockId, name) =>
  api.put(`/api/locks/${lockId}`, { name });

export const lockLock = (lockId) => api.put(`/api/locks/${lockId}/lock`);

export const unlockLock = (lockId) => api.put(`/api/locks/${lockId}/unlock`);

// ── Lock Roles ────────────────────────────────────────────────────────────────
export const getLockRoles = (lockId) => api.get(`/api/locks/${lockId}/roles`);

export const addUserToLock = (lockId, email, lockrole) =>
  api.post(`/api/locks/${lockId}/roles`, { email, lockrole });

export const editUserLockRole = (lockId, email, lockrole) =>
  api.put(`/api/locks/${lockId}/roles`, { email, lockrole });

export const deleteUserFromLock = (lockId, email) =>
  api.delete(`/api/locks/${lockId}/roles/${encodeURIComponent(email)}`);

// ── Access Keys ───────────────────────────────────────────────────────────────
export const getAllKeysOnLock = (lockId) =>
  api.get(`/api/access-key/lock/${lockId}`);

export const createAccessKey = (lockId, validFrom, validUntil) =>
  api.post(`/api/access-key/lock/${lockId}`, { validFrom, validUntil });

export const deleteAccessKey = (keyId) =>
  api.delete(`/api/access-key/${keyId}`);

File: mobile/app/src/main/java/com/hnure/smartlock/data/api/SmartLockApi.kt
package com.hnure.smartlock.data.api

import retrofit2.Response
import retrofit2.http.*

interface SmartLockApi {

    // ── Auth ─────────────────────────────────────────────────────────────────

    @POST("api/auth/login")
    suspend fun login(@Body request: LoginRequest): Response<AuthResponse>

    @POST("api/auth/register")
    suspend fun register(@Body request: RegisterRequest): Response<AuthResponse>

    // ── Locks ─────────────────────────────────────────────────────────────────

    @GET("api/locks")
    suspend fun getAllLocks(): Response<List<LockDto>>

    @POST("api/locks")
    suspend fun createLock(@Body request: CreateLockRequest): Response<LockDto>

    @DELETE("api/locks/{lockId}")
    suspend fun deleteLock(@Path("lockId") lockId: String): Response<Unit>

    @PUT("api/locks/{lockId}")
    suspend fun editLock(
        @Path("lockId") lockId: String,
        @Body request: EditLockRequest
    ): Response<LockDto>

    @PUT("api/locks/{lockId}/lock")
    suspend fun lockLock(@Path("lockId") lockId: String): Response<LockDto>

    @PUT("api/locks/{lockId}/unlock")
    suspend fun unlockLock(@Path("lockId") lockId: String): Response<LockDto>

    // ── Lock Roles ────────────────────────────────────────────────────────────

    @GET("api/locks/{lockId}/roles")
    suspend fun getLockRoles(@Path("lockId") lockId: String): Response<List<LockRoleDto>>

    @POST("api/locks/{lockId}/roles")
    suspend fun addUserToLock(
        @Path("lockId") lockId: String,
        @Body request: EditLockRoleRequest
    ): Response<LockRoleDto>

    @PUT("api/locks/{lockId}/roles")
    suspend fun editUserLockRole(
        @Path("lockId") lockId: String,
        @Body request: EditLockRoleRequest
    ): Response<LockRoleDto>

    @DELETE("api/locks/{lockId}/roles/{email}")
    suspend fun deleteUserFromLock(
        @Path("lockId") lockId: String,
        @Path("email", encoded = false) email: String
    ): Response<Unit>

    // ── Access Keys ───────────────────────────────────────────────────────────

    @GET("api/access-key/lock/{lockId}")
    suspend fun getAllKeysOnLock(@Path("lockId") lockId: String): Response<List<AccessKeyDto>>

    @POST("api/access-key/lock/{lockId}")
    suspend fun createAccessKey(
        @Path("lockId") lockId: String,
        @Body request: CreateKeyRequest
    ): Response<AccessKeyDto>

    @DELETE("api/access-key/{keyId}")
    suspend fun deleteAccessKey(@Path("keyId") keyId: String): Response<Unit>
}
